### PR TITLE
feat: Arrange link previews in messages view in a flow layout

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do

Closing: #12179 

Changes:
1. Replace ColumnLayout with Flow
2. Move the image previews up to be presented before the link previews

NOTE: Currently the user cannot send both link previews and raw images
### Affected areas

LinksMessageView

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/06a2ad3c-f83d-48d9-bad5-4a90725f2c8d

- [x] I've checked the design and this PR matches it

### CODE REVIEW

**Comment by alexjba** [comment]
@igor-sirotin Do we have a task to enable the user to send both link previews and raw images?

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 1]
```diff
@@ -27,6 +26,59 @@ ColumnLayout {
 
     signal imageClicked(var image, var mouse, var imageSource, string url)
 
+    spacing: 12
+
+    //TODO: remove once GIF previews are unfurled sender side
+    Repeater {
```
Comment (caybro) [inline ui/imports/shared/views/chat/LinksMessageView.qml:32]: Yikes, conflicts ahead

**Review by caybro — APPROVED** [review]
LGTM

But how do you get those youtube/github previews? :D

File: ui/imports/shared/views/chat/LinksMessageView.qml [hunk 2]
```diff
@@ -27,6 +26,59 @@ ColumnLayout {
...
+        delegate: Loader {
+            id: tempLoader
+
+            required property var result
+            required property string link
+            required property int index
+            required property bool unfurl
+            required property bool success
+            required property bool isStatusDeepLink
+            readonly property bool isImage: result.contentType ? result.contentType.startsWith("image/") : false
```
Comment (alexandraB99) [inline ui/imports/shared/views/chat/LinksMessageView.qml:46]: `result.contentType && result.contentType.startsWith("image/")` ?

**Review by alexandraB99 — APPROVED** [review]
just one small comment, otherwise lgtm

**Review by igor-sirotin — APPROVED** [review]

**Comment by alexjba** [comment]
> > @igor-sirotin Do we have a task to enable the user to send both link previews and raw images?
> 
> Hmmm. We don't have such task. But isn't it not currently possible? 🤔

It's not working. I've had a brief look on the API and it seems we have different status-go API to send images or regular messages

**Pull request merged**